dinstaller-cli/src/main.rs

```diff
@@ -27,91 +29,88 @@ struct Cli {
     pub format: Option<Format>,
 }
 
-async fn probe(manager: &ManagerClient<'_>) {
-    let probe = task::spawn(async {
-        // use new manager here
-        let another_manager = ManagerClient::new(dinstaller_lib::connection().await.unwrap())
-            .await
-            .unwrap();
-        another_manager.probe().await.unwrap()
-    });
-    block_on(show_progress(&manager));
+async fn probe(manager: &ManagerClient<'_>) -> Result<(), Box<dyn Error>> {
+    let another_manager = build_manager().await?;
+    let probe = task::spawn(async move { another_manager.probe().await });
+    show_progress(manager).await?;
 
-    probe.await
+    Ok(probe.await?)
 }
 
-async fn install(manager: &ManagerClient<'_>) {
-    if !manager.can_install().await.unwrap() {
+async fn install(manager: &ManagerClient<'_>) -> Result<(), Box<dyn Error>> {
```

Comment:
I expect that final goal is to not have this dyn Error and instead specify known error types like some enum, not?

Reply:
Yes, but we need to think carefully about the errors. I do not want to end up with a ["ball of mud" error enum](https://www.lpalmieri.com/posts/error-handling-rust/#avoid-ball-of-mud-error-enums) :-)

Actually, I was also considering whether we should use [anyhow](https://crates.io/crates/anyhow) to make it simpler to add context to errors.

Reply:
OK, but I hope that in the end we have simple two or three enums like DBus Error (so our side problem), User Error (user fault) and SystemError ( can be both user or our fault )